src/R/Components/Impl/ConnectionManager/Implementation/View/ConnectionManagerControl.xaml

```diff
@@ -126,116 +125,65 @@
                             <ColumnDefinition Width="*" />
                             <ColumnDefinition Width="Auto" />
                             <ColumnDefinition Width="Auto" />
+                            <ColumnDefinition Width="Auto" />
                         </Grid.ColumnDefinitions>
 
-                        <!-- Collapsed -->
-                        <Rectangle Fill="{StaticResource IconArrowRigth16}" Height="16" Width="16" />
-
-                        <!-- Connect button -->
+                        <!-- Connection icon -->
                         <ContentPresenter DataContext="{Binding}" Grid.Column="1" 
                                           ContentTemplate="{StaticResource ConnectButtonDataTemplate}" />
 
-                        <!-- Name & path -->
+                        <!-- Name, path, command line  -->
                         <TextBlock Grid.Column="2" Margin="0,1,6,2" Text="{Binding Path=Name}"/>
-                        <Separator Grid.Column="3" Margin="0,1,6,2" Style="{StaticResource VerticalSeparator}"/>
+                        <Separator Grid.Column="3" Margin="0,1,6,2" Style="{StaticResource VerticalSeparator}"
+                                   Visibility="{Binding Path=IsRemote, Converter={x:Static rwpf:Converters.FalseIsCollapsed}}"/>
                         <TextBlock Grid.Column="4" Margin="0,1,3,2" Text="{Binding Path=Path, Mode=OneWay}"
-                                   TextTrimming="CharacterEllipsis" TextWrapping="NoWrap"/>
+                                   TextTrimming="CharacterEllipsis" TextWrapping="NoWrap"
+                                   Visibility="{Binding Path=IsRemote, Converter={x:Static rwpf:Converters.FalseIsCollapsed}}"/>
+
+                        <!-- connect button -->
+                        <Button x:Name="ButtonConnect" Grid.Column="5" Style="{StaticResource ToolWindowButtonStyle}" Click="ButtonConnect_Click" 
+                                Visibility="{Binding Path=IsConnected, Converter={x:Static rwpf:Converters.TrueIsCollapsed}}"
+                                Padding="0,0,8,0">
```

Comment:
Then let's use `Margin`, cause `Padding` may not offset the background, which is also considered part of the button

---
In reply to: [77670185](https://github.com/Microsoft/RTVS/pull/2246#discussion_r77670185) [](ancestors = 77670185)